server/images.go

```diff
@@ -115,6 +116,24 @@ func (m *Model) Capabilities() []model.Capability {
 		capabilities = append(capabilities, model.CapabilityThinking)
 	}
 
+	// Check for reranking capability - enhanced detection
+	vars := m.Template.Vars()
+	hasRerankVars := slices.Contains(vars, "query") && slices.Contains(vars, "document")
+	
+	if hasRerankVars {
+		// Check template content for reranking patterns
```

Comment:
Since the query and document are newly added for rerank, is the following double check necessary